src/pages/ServiceDetails/ServiceDetailsPage.tsx

```diff
@@ -428,12 +429,13 @@ class ServiceDetails extends React.Component<ServiceDetailsProps, ServiceDetails
       tabsArray.push(jaegerTag);
     }
 
-    if (this.state.serviceDetailsInfo.apiDocumentation && this.state.serviceDetailsInfo.apiDocumentation.hasSpec) {
+    if (this.state.serviceDetailsInfo.apiDocumentation && this.state.serviceDetailsInfo.apiDocumentation.path !=='') {
```

Comment:
Because server-side, the path is always set to `/` at minimum, it's never empty, so the API Doc tab is always displayed even when the annotations are not present.
(I guess that can be fixed server-side)

Reply:
The path here is the path from the service annotation, not the general path of Kiali, so it should be empty when not filled by the service

@israel-hdez I rebased the branch

Reply:
@couniojc I know it's taken from the annotation, but see here: https://github.com/kiali/kiali/pull/1815/files#diff-b8ee2df445e5c9205357eb4bc9419746R236-R239 : an empty path would be transformed to `/`